File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/zig/F17.P2.zig
const std = @import("std");

var released: i32 = 0;

fn probe() i32 {
// BEGIN PROBE F17.P2
const Handle = struct {
    id: i32,
    fn deinit(_: @This()) void {
        released += 1;
    }
};

{
    const h = Handle{ .id = 1 };
    defer h.deinit();
}

const result = released;
// END PROBE F17.P2
    return result;
}

pub fn main() !void {
    var threaded: std.Io.Threaded = .init_single_threaded;
    const io = threaded.io();
    var out_buf: [64]u8 = undefined;
    var out = std.Io.File.stdout().writer(io, &out_buf);
    try out.interface.print("{d}\n", .{probe()});
    try out.interface.flush();
}

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/swift/F17.P2.swift
var released: Int32 = 0

// BEGIN PROBE F17.P2
class Handle {
    let id: Int32
    init(_ id: Int32) { self.id = id }
    deinit { released += 1 }
}
do {
    let h = Handle(1)
}
let result = released
// END PROBE F17.P2

print(result)
